### ISSUE TITLE
[Bug]: JIT: media break points do not generate for postcss7-compat

### ISSUE THREAD

**Issue by gcavanunez** [tailwindlabs/tailwindcss#4680]
### What version of Tailwind CSS are you using?

v2.2.0

### What build tool (or framework if it abstracts the build tool) are you using?

@vue/cli 4.5.12 | postcss 7

### What version of Node.js are you using?

v15.6.0

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### Reproduction repository

https://github.com/gcavanunez/tailwindcss-jit-prefix-vue-sample

### Describe your issue

Hello, first of congrats on the new release, tons of awesome features I can't imagine shipping the features can be simple.

Describe the bug
When enabling the JIT mode media queries are not generated properly. 

Additional context
I initially found the issue while working with a prefix key then to make sure if it was an issue only with the prefix key activated I disabled it and ran it against the Base.vue file that's a copy of the play.tailwindcss.com.

I'm using postcss to compile to further isolate the issue, I tried using the tailwind cli but since it requires postcss 8 to work was unable to use it, some of the code from another example component I've extracted contains vuetify components, but bringing it on board to demonstrate the issue is not necessary. 
To Reproduce

1. yarn dev-tw-sample
2. view the tw.build.css under src directory

Screenshots
![image](https://user-images.githubusercontent.com/9332696/122447558-7bc5ac00-cf69-11eb-923a-9c9778b98529.png)

Issues/Pull requests that touch on the issue
https://github.com/tailwindlabs/tailwindcss/pull/4078
https://github.com/tailwindlabs/tailwindcss/issues/4151
https://github.com/tailwindlabs/tailwindcss/issues/4034#issuecomment-817332062

**Comment by charklewis** [comment]
I have also found that when enabling dark mode, the dark classes are always applied. I have followed to use tailwind exactly as in the [create-react-app](https://tailwindcss.com/docs/guides/create-react-app) guide.

My config looks like:

```
const colors = require("tailwindcss/colors")

module.exports = {
  mode: "jit",
  purge: ["./src/**/*.{js,jsx,ts,tsx}", "./public/index.html"],
  darkMode: "media",
  theme: {
    extend: {
      colors: {
        transparent: "transparent",
        current: "currentColor",
        black: colors.black,
        white: colors.white,
        gray: colors.trueGray,
        indigo: colors.indigo,
        red: colors.rose,
        yellow: colors.amber
      }
    }
  },
  variants: {
    extend: {
      margin: ["first", "last"],
      opacity: ["disabled"],
      backgroundColor: ["disabled"],
      textColor: ["disabled"],
      cursor: ["disabled"],
      padding: ["disabled"],
      display: ["group-hover"]
    }
  },
  plugins: [require("@tailwindcss/forms")]
}
```

**Comment by amaelftah** [comment]
i have the same issue too

**Comment by w99910** [comment]
I'm having the same issue too.

**Comment by imjohnlouie04** [comment]
media query got break to my project too

**Comment by adamwathan** [comment]
Should be fixed in v2.2.1! 🥳

**Comment by imjohnlouie04** [comment]
> Should be fixed in v2.2.1! 🥳

Got tested on my project and viola working properly.. love it

**Comment by maximeg** [comment]
Nickel !

Thanks
